### PR TITLE
Delete index.html + clean up initialization

### PR DESCRIPTION
**Pull request by marcoscaceres**
- Removed index.html. 
- Adjust URLs relative to new SW location
- Cleaned up how initialization happens a bit.

The use of gWhatever is a real disaster :( There is so much co-dependent code that doing initialization is currently a nightmare. In the future, we need to get rid of all interdependent code - i.e., only use events instead of globals.

### CODE REVIEW

**Comment by marcoscaceres** [comment]
@emtwo, @oyiptong, @sarracini, r?

File: bin/generate-offline-files.js [hunk 1]
```diff
@@ -22,9 +22,9 @@ function processResults(results) {
     .reduce((current, next) => next.concat(current), [])
     .split("\n")
     .filter(item => item)
-    .map(item => item.replace(/^src\//, "/"))
+    .map(item => item.replace(/^src\//, "./"))
```
Comment (marcoscaceres) [inline bin/generate-offline-files.js:25]: The Service Worker now lives ones directory lower than the index.html (as do all the JS, CSS, SVG resources). Hence, "load relative to the SW".

File: bin/generate-offline-files.js [hunk 2]
```diff
@@ -22,9 +22,9 @@ function processResults(results) {
...
     .sort();
-  return JSON.stringify(paths.concat(["/"]), null, 2);
```
Comment (marcoscaceres) [inline bin/generate-offline-files.js:27]: We can't load the "base" any more this way, the index.html file now handles its own storage... see `addIndexToCache()` task below.

File: src/js/newTab.js [hunk 3]
```diff
@@ -1,27 +1,61 @@
...
 
 (function(exports) {
   const gNewTab = {
 
-    listeners: {},
+    listeners: new Map(),
 
     _l10nStrings: new Map(),
 
+    privateBrowsingMode: false,
```
Comment (marcoscaceres) [inline src/js/newTab.js:15]: I made sure these properties were no longer added dynamically. Adding props dynamically to objects can be very confusing.

File: src/js/newTab.js [hunk 4]
```diff
@@ -1,27 +1,61 @@
...
+
+    introShown: false,
+
+    windowID: -1,
+
+    enabled: true,
+
     init() {
       return async.task(function*() {
+        yield this._cacheL10nStrings();
```
Comment (marcoscaceres) [inline src/js/newTab.js:31]: I made initialization of gNewTab self contained here.

File: src/js/newTab.js [hunk 5]
```diff
@@ -69,7 +101,7 @@
       }
     },
 
-    setInitialState(message) {
+    _setInitialState(message) {
```
Comment (marcoscaceres) [inline src/js/newTab.js:106]: I couldn't see any good reason for this to be public, so I switched it to private.

**Comment by oyiptong** [comment]
Can we move the initialization inline scripts in index.html into an init.js script?
It cleans up the index page, diminishes the number of bytes per page, and clears a requirement from content security (for meta-csp, to not have any inline scripts or styles).

File: src/js/newTab.js [hunk 6]
```diff
@@ -1,27 +1,61 @@
...
+
+    rows: -1,
+
+    columns: -1,
+
+    introShown: false,
+
+    windowID: -1,
+
+    enabled: true,
```
Comment (sarracini) [inline src/js/newTab.js:25]: we'll also need an `enhanced` here i believe

Comment (marcoscaceres) [inline src/js/newTab.js:25]: Added

**Comment by marcoscaceres** [comment]
> Can we move the initialization inline scripts in index.html into an init.js script?

On it!

**Comment by marcoscaceres** [comment]
@oyiptong, added init.js.

**Comment by marcoscaceres** [comment]
@oyiptong r?

**Comment by oyiptong** [comment]
r+

**Pull request merged**